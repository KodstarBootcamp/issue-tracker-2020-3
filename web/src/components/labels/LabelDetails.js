import React,{ useState } from 'react'
import { LabelEditForm } from './LabelEditForm'
import { BsTrash, BsPencil } from 'react-icons/bs'

export const LabelDetails = ( props ) => {
  const [viewLabelEdit,setViewLabelEdit] = useState(false)

  return (
    <tr>
      <td>
        {viewLabelEdit&&
          <LabelEditForm key={props.label.id} label={props.label} setLabels={props.setLabels} setIssues={props.setIssues}
            setInfoMessage={props.setInfoMessage} setViewLabelEdit={setViewLabelEdit}/>}
        {!viewLabelEdit&&<>{props.label.text}</>}
      </td>
      <td style={{ color: '#fcf8f8', backgroundColor: props.label.color }}>{props.label.color} </td>
      {props.user&&<td ><BsPencil  onClick={() => setViewLabelEdit(true)} style={{ color: 'blue' }} className="ml-4" size={16} /></td>}
      {props.user&&<td><BsTrash style={{ color: 'red' }} onClick={ () => props.handleDelete(props.label.id)} className="ml-1" /></td>}
    </tr>
  )
}
