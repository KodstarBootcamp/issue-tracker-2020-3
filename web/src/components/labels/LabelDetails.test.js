import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import { LabelDetails } from './LabelDetails'


describe('<Label test', () => {
  const LabelTest = {
    id: '5a422a851b54a676234d17f7',
    title: 'Tomas adventure',
    color: 'Jahnsen',
  }
  test('renders content', () => {

    const component = render( <LabelDetails label={LabelTest} /> )

    const element1 = component.getByText('Jahnsen')

    expect(element1).toBeDefined()
  })//end of the issue view test describe
})//end of the begining describe