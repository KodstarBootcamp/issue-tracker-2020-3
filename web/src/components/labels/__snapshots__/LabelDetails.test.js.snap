// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label test renders content 1`] = `
<tr>
  <td />
  <td
    style={
      Object {
        "backgroundColor": "Jahnsen",
        "color": "#fcf8f8",
      }
    }
  >
    Jahnsen
     
  </td>
  
  
</tr>
`;
